notebooks/_06-Architecture.md

```diff
@@ -21,398 +21,263 @@ jupyter:
-
-
-
-
+See above
 
 ### 6.5 Why are we using the particular algorithms we’ve chosen instead of others? 
-
-
-
-
+See above
 
 ### 6.6 Why do we believe the algorithms can be implemented?
-
-
-
-
+See above
 
 ### 6.7 Do we have any concerns with regard to algorithmic efficiency?
-
-
-
+See above
 
 
 ## 7. Description of data usage in the system:
-
-
 ### 7.1 How will data be ingested / accessed?
-
-
-
-
+Data will be ingested using pandas from either an online path to the dataset or from a local path on the user's machine
 
 ### 7.2 How will data be validated?
-
-
-
-
+Data can be validated using third party dependencies such
 
 ### 7.3 How will data be organized?
-
-
-
-
+Data will be organized in tabular format (.csv)
 
 ### 7.4 How will data be transformed?
-
-
-
-
+Data will be transformed using methods available within Pandas
 
 ### 7.5 How will created data be output and stored?
+Causal graphs will be output to .png files and pushed to the repo
 
+Summaries of model estimate will be saved as markdown tables
 
-
-
+Plots of results will be saved as .png files and pushed to the repo
 
 ### 7.6 How will created data be validated?
-
-
-
-
+There are packages that will validate data.
 
 ### 7.7 How does all of the above differ online vs offline? (e.g. we may throw a critical error offline to prevent training with bad data but only log a warning online to note that the data in a request was unexpected.)
-
-
-
-
+N/A
 
 ### 7.8 How will the flow / manipulation of data be controlled (and recorded / version-controlled) and made reproducible? E.g. data ingestion and manipulation to create training / testing sets.
 - **Flyte?**
 - **Make?**
 - **Other?**  
-
-
-
-
+We don't have any plans for this. please make any suggestions as you see needed/fit.
 
 ## 8. Description of experimentation protocols (if relevant):
 **If experiments need to be run (e.g. hyper-parameter tuning, model-selection, A/B tests, etc.):**
-
-
 ### 8.1 How will experiments be launched?
-
-
-
-
+N/A
 
 ### 8.2 How will experiment reproducibility be ensured?
-
-
-
-
+N/A
 
 ### 8.3 How will experiment meta-data (e.g. launch configurations) be stored?
-
-
-
-
+N/A
 
 ### 8.4 How will experiment-created data (e.g. results) be stored?
-
-
-
-
+N/A
 
 ### 8.5 How will experiment-created data be analyzed?
-
-
-
-
+N/A
 
 ### 8.6 How will experiment analyses be prepared for public reporting? 
-
-
-
-
+N/A
 
 ## 9. Description of user-interface:
-
-
 ### 9.1 How will users (including myself) interact with the system that is built?
 - **Will there be a command line interface?**
 - **Will users be editing configuration files, and if so, where will they be stored?**
 - **Will there be ability to replay the DAG representing one’s data analysis?**
 
+Users will use a jupyter notebook to interact with the system that is build
 
-
-
+No configuration files will need to be edited. If any functions need to be edited, one could easily make the adjustment
```

Comment:
I strongly advise against this.

At the very least, all parameters for the notebook should be defined in a cell at the top (the idea is the notebooks in their final state should at least be runnable via the papermill package).

Not allowing the code for the project to be run outside the notebook is highly limiting if you or someone else wants to reuse a notebook as part of a pipeline elsewhere.

We should really strive to make life as easy as possible for others, especially if it doesn't require us going to a heroic extent (like changing how we organize a notebook). Just my suggestion!

Reply:
Ah! I got what you mean now. My interpretation of configuration files was project environment configuration files (requirements.txt for example). Sorry for misunderstanding.

I think the config file should contain data relevant to any arguments or parameters (including dataset urls) that are to be used in the project. I will update the answer accordingly.

Reply:
No worries, I'm learning how to communicate more clearly about these things!